shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptProjectionTokenGenerator.java

```diff
@@ -67,63 +71,96 @@ protected boolean isGenerateSQLTokenForEncrypt(final SQLStatementContext sqlStat
+                SubqueryExtractUtil.getSubqueryTableSegmentsFromTableSegment(selectStatementContext.getSqlStatement().getFrom()).forEach(each -> result.addAll(generateSQLTokens(
+                        new SelectStatementContext(selectStatementContext.getMetaDataMap(), selectStatementContext.getParameters(), each.getSubquery().getSelect(),
+                                selectStatementContext.getSchemaName()), each.getAlias(), SubqueryEnum.NestedProjectionTabsegmentSubquery, rewriteMetaDataMap)));
+                SubqueryExtractUtil.getSubquerySegmentsFromProjections(selectStatementContext.getSqlStatement().getProjections()).forEach(each -> result.addAll(generateSQLTokens(
+                        new SelectStatementContext(selectStatementContext.getMetaDataMap(), selectStatementContext.getParameters(), each.getSelect(),
+                                selectStatementContext.getSchemaName()), Optional.empty(), SubqueryEnum.ProjectionSubqery, rewriteMetaDataMap)));
+                selectStatementContext.getWhere().ifPresent(where -> SubqueryExtractUtil.getSubquerySegmentsFromExpression(where.getExpr()).forEach(each -> result.addAll(generateSQLTokens(
+                        new SelectStatementContext(selectStatementContext.getMetaDataMap(), selectStatementContext.getParameters(), each.getSelect(),
+                                selectStatementContext.getSchemaName()), Optional.empty(), SubqueryEnum.ExpressionSubqery, rewriteMetaDataMap))));
+            }
+            result.addAll(generateSQLTokens(selectStatementContext, Optional.empty(), SubqueryEnum.None, rewriteMetaDataMap));
         }
         return result;
     }
-    
-    private Collection<SubstitutableColumnNameToken> generateSQLTokens(final SelectStatementContext selectStatementContext, final boolean insertSelect) {
+
+    private Collection<SubstitutableColumnNameToken> generateSQLTokens(final SelectStatementContext selectStatementContext, final Optional<String> alias,
+                                                                       final SubqueryEnum subqueryEnum, final Map<String, Map<String, Map<String, String>>> rewriteMetaDataMap) {
         Collection<SubstitutableColumnNameToken> result = new LinkedHashSet<>();
         ProjectionsSegment projectionsSegment = selectStatementContext.getSqlStatement().getProjections();
         for (String each : selectStatementContext.getTablesContext().getTableNames()) {
-            getEncryptRule().findEncryptTable(each).map(optional -> generateSQLTokens(projectionsSegment, each, selectStatementContext, optional, insertSelect)).ifPresent(result::addAll);
+            getEncryptRule().findEncryptTable(each).map(optional -> generateSQLTokens(projectionsSegment, each, selectStatementContext, optional, alias, subqueryEnum,
+                    rewriteMetaDataMap)).ifPresent(result::addAll);
         }
+        selectStatementContext.setRewriteMetaDataMap(rewriteMetaDataMap);
```

Comment:
Why set rewriteMetaDataMap to Context? The value in the Context should be completed when the Context is initialized.

Reply:
set rewriteMetaDataMap and get it to rewrite related column in Projections and Expressions.